### ISSUE TITLE
amd-function-arity warns when first argument to require/define is not an array literal

### ISSUE THREAD

**Issue by platinumazure** [cvisco/eslint-plugin-requirejs#50]
Test case:

``` js
require(that._configModel.getDependenciesPath(), function (product) {
    that._dependenciesLoaded(product);
});
```

Expected: No warning (how can it know what the function call will return?).

Actual: "Too many parameters in require callback (expected 0, found 1)."

Presumably it should just bail out without warning if the first argument to `define`/`require`/`requirejs` is not an ArrayExpression.

If this is accepted as a bug, I can work on a fix.

**Comment by cvisco** [comment]
@platinumazure, agreed. If it's not an ArrayExpression, it should not warn at all.
